src/sentry/monitors/logic/mark_failed.py

```diff
@@ -175,22 +173,30 @@ def mark_failed_no_threshold(
         pass
 
     if use_issue_platform:
-        create_issue_platform_occurrence(monitor_env, reason, occurrence_context)
+        create_issue_platform_occurrence(failed_checkin)
     else:
-        create_legacy_event(monitor_env, reason)
+        create_legacy_event(failed_checkin)
 
     monitor_environment_failed.send(monitor_environment=monitor_env, sender=type(monitor_env))
 
     return True
 
 
-def create_legacy_event(monitor_env: MonitorEnvironment, reason: str):
+def create_legacy_event(failed_checkin: MonitorCheckIn):
     from sentry.coreapi import insert_data_to_database_legacy
     from sentry.event_manager import EventManager
     from sentry.models import Project
 
+    monitor_env = failed_checkin.monitor_environment
     context = get_monitor_environment_context(monitor_env)
 
+    # XXX(epurkhiser): This matches up with the occurrence_data reason
+    reason_map = {
+        CheckInStatus.MISSED: "missed_checkin",
+        CheckInStatus.TIMEOUT: "duration",
+    }
```

Comment:
put this up top as a constant?

Reply:
I think it's fine, we don't need it in any of the other functions and I would rather it not be "exported"